### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by NasKhalil**
- Create unit tests for all React components:
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.

### CODE REVIEW

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      total: 20,
+      next: 10,
+      operation: '-',
+    };
+    const buttonName = '=';
+    expect(calculate(obj, buttonName)).toStrictEqual({
+      total: '10',
+      next: null,
+      operation: null,
+    });
```
Comment (D4nielJ) [inline src/tests/calculate.test.js:15]: - [x] Please cover all the functionalities and scenarios of the calculate function. Or as much as you can think of. (Every operation, forbidden operations as division or modulus by 0, what happens if strings are sent? etc.

Making sure to cover every scenario when making tests is like creating a layer of protection to our code. If later on, we want to implement new functionalities to the calculator, we will know we're doing something wrong if out of a sudden tests start failing.

File: src/tests/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import operate from '../logic/operate';
+
+describe('mathematical operations', () => {
+  test('Add two numbers', () => {
+    expect(operate(5, 10, '+')).toBe('15');
+  });
+
+  test('Subtract two numbers', () => {
+    expect(operate(20, 10, '-')).toBe('10');
+  });
```
Comment (D4nielJ) [inline src/tests/operate.test.js:10]: - [x] As with the calculate function, please cover all the functionalities and scenarios of the `operate` function. Or as much as you can think of. (Every operation, forbidden operations as division or modulus by 0, what happens if strings are sent? etc.

File: src/tests/Home.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Home from '../pages/Home';
+
+it('renders correctly', () => {
+  const tree = renderer
+    .create(<Home />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (D4nielJ) [inline src/tests/Home.test.js:10]: - [x] Please add integration tests as specified on the requirements of this project.

> There should be the usage of React Testing Library to simulate user interaction.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @NasKhalil and @zemola, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: 
- Current tests are well structured and test functionality correctly!
- Good job creating snapshots. :rocket: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Navbar.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (SajjadAhmad14) [inline src/tests/Navbar.test.js:1]: Please add integration tests using React Testing Library as it's requested by previous code reviewer

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**